e2etest/pkg/frr/container/container.go

```diff
@@ -19,20 +20,78 @@ const (
 	frrConfigDir = "config/frr"
 	// FRR routing image.
 	frrImage = "frrouting/frr:v7.5.1"
+	// Host network name.
+	hostNetwork = "host"
 	// FRR container mount destination path.
 	frrMountPath = "/etc/frr"
 )
 
+type Frr struct {
+	executor.Executor
+	Name           string
+	configDir      string
+	NeighborConfig config.NeighborConfig
+	RouterConfig   config.RouterConfig
+	Ipv4           string
+	Ipv6           string
+}
+
+// New creates a new FRR container on the host and returns the corresponding *Frr.
+// A situation where a non-nil container and an error are returned is possible.
+func New(name string, nc config.NeighborConfig, rc config.RouterConfig, network string, hostipv4 string, hostipv6 string) (*Frr, error) {
+	configDir, err := ioutil.TempDir("", "frr-conf")
+	if err != nil {
+		return nil, err
+	}
+
+	err = startContainer(name, configDir, rc, network)
+	if err != nil {
+		return nil, err
+	}
+
+	exc := executor.ForContainer(name)
+
+	frr := &Frr{
+		Executor:       exc,
+		Name:           name,
+		configDir:      configDir,
+		NeighborConfig: nc,
+		RouterConfig:   rc,
+	}
+
+	if network == hostNetwork {
+		frr.Ipv4 = hostipv4
+		frr.Ipv6 = hostipv6
+	} else {
+		err = frr.updateIPS()
+		if err != nil {
+			return frr, err
+		}
+	}
+
+	err = frr.updateVolumePermissions()
+	if err != nil {
+		return frr, err
+	}
+
+	return frr, nil
+}
+
 // Run a BGP router in a container.
-func StartContainer(containerName string, testDirName string) error {
+func startContainer(containerName string, testDirName string, rc config.RouterConfig, network string) error {
 	srcFiles := fmt.Sprintf("%s/.", frrConfigDir)
 	res, err := exec.Command("cp", "-r", srcFiles, testDirName).CombinedOutput()
 	if err != nil {
 		return errors.Wrapf(err, "Failed to copy FRR config directory. %s", string(res))
 	}
 
+	err = config.SetDaemonsConfig(testDirName, rc)
+	if err != nil {
+		return err
+	}
+
 	volume := fmt.Sprintf("%s:%s", testDirName, frrMountPath)
-	out, err := exec.Command("docker", "run", "-d", "--privileged", "--network", "kind", "--rm", "--name", containerName,
+	out, err := exec.Command("docker", "run", "-d", "--privileged", "--network", network, "--rm", "--name", containerName,
```

Comment:
why do we need to run as privileged ?

Reply:
that wasn't part of my commit :sweat_smile:

Reply:
probably because the routes are added to kernel however with view usage we don't install routes in the kernel maybe we can remove that later i will check it out

Reply:
The external containers that act as bgp peers are not related to how we configure FRR as a speaker (as a different BGP mode). Using views in the external containers would add up complexity because we'll need to find different ways to ping/wget/... since the routes won't be installed in the container.

Reply:
well I was thinking we can use views in speaker configs and we can have two frr external containers ?